refactor(core): WIP - partial hydration prototype

thePunderWoman opened the pull request:
This is a work in progress of the prototype of partial hydration using defer blocks

bencodezen committed:
docs: resolve extra padding on tutorial editor (#54828)

As of this commit, there is a visual discrepancy between the height of the content on the left compared to the separator and editors on the right. It appears that the cause of this is an additional computation of layout padding which is unnecessary for these particular elements.

PR Close #54828

crisbeto committed:
refactor(compiler-cli): add compiler flag for testing let declarations (#56199)

Adds a private `_enableLetSyntax` flag that allows for let declarations to be enabled in tests.

PR Close #56199

crisbeto committed:
refactor(compiler): integrate let declarations into the template binder (#56199)

Integrates the let declarations into the template binder which will be used to power the scoping behavior of the new syntax.

PR Close #56199

crisbeto committed:
refactor(compiler-cli): integrate let declarations into the template type checker (#56199)

Integrates let declarations into the template type checker by producing corresponding constants in the TCB.

This also includes a couple of custom diagnostics to flag usages of let before they're declared and illegal writes to let declarations. We can't rely on TS for these checks, because it includes the variable name in the diagnostic.

PR Close #56199

crisbeto committed:
refactor(compiler-cli): integrate let declaration into the indexer (#56199)

Adds support for let declarations in the template indexer.

PR Close #56199

crisbeto committed:
refactor(compiler-cli): support resolving the symbol of let declaration (#56199)

Updates the symbol builder to handle resolving the symbol of a let declaration.

PR Close #56199

crisbeto committed:
refactor(compiler-cli): support completions for let declarations (#56199)

Adds support for let declarations inside the `CompletionEngine`.

PR Close #56199

bencodezen committed:
docs: fix vertical playground spacing issue (#55689)

PR Close #55689

thePunderWoman committed:
refactor(core): WIP - partial hydration prototype

This is a work in progress of the prototype of partial hydration using defer blocks

dylhunn committed:
docs(zone.js): update release guide for zone.js (#55846)

Update the releasing guide for zone.js, because one of the release commands was outdated.
PR Close #55846

9kubczas4 committed:
fix(docs-infra): remove config release from test scripts (#56062)

fix(docs-infra): remove config release from test scripts

As we discussed in one of the previous PRs we should remove `--config=release` from test scripts on CI
fix(docs-infra): use DI to inject current VERSION.major

PR Close #56062

arturovt committed:
fix(zone.js): store remove abort listener on the scheduled task (#56160)

Prior to this commit, a memory leak occurred when the `abort` listener was
not removed from the `AbortSignal`. We introduced a fix to remove the event
listener, but it was erroneously stored on the `taskData`, which is a shared
global object. Consequently, when something attempted to remove an event listener,
it immediately removed the last stored abort listener. As a result, events would
never be canceled.

We have now rectified this by storing the remove abort listener function directly on
the task itself. This adjustment ensures that the abort listener is tied only to the
specific task. When the `abort` function is called, it cancels the task. Therefore, it
is safe to associate the cleanup function directly with the task.

Closes: #56148

PR Close #56160

iteriani committed:
refactor(core): Refactor parts of event_replay into a shared library that will be used with global event delegation. (#56172)

This also moves the code that stashes the jsaction more closely to the code that actually sets the event listener.

PR Close #56172

alan-agius4 committed:
build: update Node.js to match Angular CLI engines (#56187)

The current supported Node.js engines by the Angular CLI are `^18.19.1 || ^20.11.1 || >=22.0.0`

PR Close #56187

tbondwilkinson committed:
refactor(core): Add and move tests for `Dispatcher`. (#56193)

Move tests from `eventcontract_test.ts` to `dispatcher_test.ts` for
functionality that lives in `Dispatcher`.

Add an extra test for `preventDefault` behavior for `CLICKMOD` that
covers a previous bug case.

PR Close #56193

angular-robot committed:
build: update io_bazel_rules_sass digest to 61dde52 (#56168)

See associated pull request for more information.

PR Close #56168

crisbeto committed:
refactor(compiler-cli): account for let declarations in two-way binding check (#56199)

Updates the check that prevent writes to template variables in two-way bindings to account for let declarations.

Also fixes some old tests that weren't properly setting up two-way bindings.

PR Close #56199

iteriani commented on packages/core/src/hydration/event_replay.ts:
```diff
@@ -185,23 +194,60 @@ export function setJSActionAttribute(
     const events = nativeElementToEvents.get(nativeElement) ?? [];
-    const parts = events.map((event) => `${event}:`);
+    const namespace = parentDeferBlockId ?? '';
+    const parts = events.map((event) => `${event}:${namespace}`);
```
would it make sense to serialize on a separate attribute so we don't repeat?

thePunderWoman replied:
WDYM? What would be repeated?

iteriani replied:
I was thikning instead of

jsaction="click:d1;focus:d1"

it'd be

jsaction-block="d1" jsaction="click;"

then we wouldn't have to regex

thePunderWoman replied:
It would be more bites over the wire to do that with the additional long attribute. Is there a desire to not use the action portion of the jsaction string? We also don't have to regex. I just did it that way because it was easy enough to do. I could do it via string splitting, too.

iteriani replied:
It's more that we do this work here as well (https://github.com/angular/angular/blob/01172b84d9e8ab37ea989783a480add80c442421/packages/core/primitives/event-dispatch/src/action_resolver.ts#L247) and it'd be nice to share that work (especially since it's cached here).

iteriani commented on packages/core/src/hydration/event_replay.ts:
```diff
@@ -185,23 +194,60 @@ export function setJSActionAttribute(
+async function handleEvent(
+  appRef: ApplicationRef,
+  event: Event,
+  blockName: string,
```
can we make the latter two fields optional to avoid the set and closure allocation?

thePunderWoman replied:
Certainly

JeanMeche committed:
refactor(core): Use Nullish coalescing assignment for view getters. (#56242)

Micro optim, this wasn't minified by the optimizer.

PR Close #56242

angular-robot committed:
build: update cross-repo angular dependencies (#56259)

See associated pull request for more information.

Closes #56184 as a pr takeover

PR Close #56259

alvarocjunq committed:
docs: fixing typo `:ng-deep` to `::ng-deep` (#56251)
PR Close #56251

crisbeto committed:
refactor(compiler-cli): add diagnostic for duplicate let declarations (#56199)

Adds a template diagnostic that will flag cases where multiple `@let` declarations use the same name.

PR Close #56199

pkozlowski-opensource committed:
refactor(docs-infra): mark the update component as OnPush (#56254)

After inspecting and testing the adev-update-guide component it
seems like it could be marked as OnPush. This change gets us closer
to enabling zoneless for adev.

PR Close #56254

pkozlowski-opensource committed:
refactor(docs-infra): mark code editors components as OnPush (#56254)

After inspecting and testing the embedded-editor and adev-code-editor
components it
seems like those could be marked as OnPush.
This change gets us closer to enabling zoneless for adev.

PR Close #56254

pkozlowski-opensource committed:
refactor(docs-infra): mark adev-root component as OnPush (#56254)

After inspecting and testing the adev-root component is seems like
it could be marked as OnPush.
This change gets us closer to enabling zoneless for adev.

PR Close #56254

sasidharansd committed:
docs: fix docs links (#56234)

PR Close #56234

crisbeto committed:
refactor(language-service): integrate let declarations (#56270)

Integrates let declarations in the various places within the language service (quick info, completions etc).

PR Close #56270

MarkTechson committed:
docs: remove v18 developer event banner (#56278)

PR Close #56278

angular-robot committed:
build: update dependency @rollup/plugin-commonjs to v26 (#56281)

See associated pull request for more information.

PR Close #56281

thePunderWoman committed:
docs: release notes for the v18.0.2 release

thePunderWoman committed:
release: cut the v18.1.0-next.1 release
